File: 이지수/Swift/Baekjoon_15624.swift
import Foundation
func fibo(_ n: Int) -> Int {
    guard n > 1 else { return n == 1 ? 1 : 0 } 
    var a: Int = 0
    var b: Int = 1
    for _ in 0..<(n-1) {
        let temp = b
        b = (a+b) % 1000000007
        a = temp
    }
    return b
}
print(fibo(Int(readLine()!)!))

File: 이지수/Python/Baekjoon_15624.py
import sys
input = sys.stdin.readline

n = int(input().strip())
def fibo(n) :
    a, b = 0, 1
    for _ in range(n-1) : a, b = b, (a+b)%1000000007
    return b
if n < 2 : print(n)
else : print(fibo(n))
